esmvalcore/preprocessor/_regrid.py

```diff
@@ -384,6 +384,18 @@ def extract_location(cube, location, scheme):
                          scheme)
 
 
+def _check_grid_discontiguities(cube, scheme):
+    """Check if there are grid discontiguities and set use_src_mask to True."""
```

Comment:
```suggestion
    """Check if there are grid discontiguities and set use_src_mask to True."""
    scheme = dict(scheme)
```
to avoid updating the input `scheme` dict? It looks like the intention of the function is to return a new `scheme` dict, but currently it modifies the input dict.